DataFormats/simulation/src/Stack.cxx

```diff
@@ -40,7 +39,7 @@ Stack::Stack(Int_t size)
     mStack(),
     //mParticles(new TClonesArray("TParticle", size)),
     mParticles(),
-	mTracks(new std::vector<o2::MCTrack>),
+    mTracks(new std::vector<o2::MCTrack>),
```

Comment:
actually, what's the reason to have a pointer to vector instead of a vector?

Reply:
This comment is however not related to the development proposed in this PR; in the change above, I have just replace tabulation with spaces to fix the indention.

Nevertheless, I tried to replace the pointer data member with a vector, and changed the call in Register() to:
  auto* ptrmTracks = &mTracks;
  FairRootManager::Instance()->RegisterAny("MCTrack", ptrmTracks, kTRUE);
as calling just
  FairRootManager::Instance()->RegisterAny("MCTrack",&mTracks, kTRUE);
does not compile,
but the program breaks.
The same pattern is used also for registration of hits collections in Detector, which are also define via new. So it should be better reviewed by Sandro who implemented this new registration method in FairRoot.

The assignment operator is declared private, and it is not used by the class itself, so it can be declared as deleted. On the other hand this breaks a usual symmetry with copy constructor which has to be provided as it is used by CloneStack() function.

Reply:
Currently the pointer is still required.

Reply:
thanks Ivana for checking the case. sorry for the noise